### PR TITLE
tests: make several txn tests less flaky

### LINKED ISSUE #12520: Flaky integration tests `test_atomic_getting_max_ts_and_storing_memory_lock` and `test_pessimistic_lock_check_epoch`
From CI:
https://ci.pingcap.net/blue/rest/organizations/jenkins/pipelines/tikv_ghpr_test/runs/20271/nodes/86/steps/173/log/?start=0

### PR DESCRIPTION
**Pull request by sticnarf**
### What is changed and how it works?

Issue Number: Close #12520

What's Changed:

```commit-message
This commit intends to make test_atomic_getting_max_ts_and_storing_memory_lock
and test_pessimistic_lock_check_epoch less flaky.

Instead of using pause and sleep to control order, this commit uses channels
and fail::callback_cfg to control the order more deterministically.
```

### Check List 

Tests 

- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by sticnarf** [comment]
cc @tonyxuqqi

File: tests/failpoints/cases/test_transaction.rs [hunk 1]
```diff
@@ -68,8 +71,14 @@ fn test_atomic_getting_max_ts_and_storing_memory_lock() {
         .unwrap();
 
     let (prewrite_tx, prewrite_rx) = channel();
+    let (fp_tx, fp_rx) = sync_channel(1);
     // sleep a while between getting max ts and store the lock in memory
-    fail::cfg("before-set-lock-in-memory", "sleep(500)").unwrap();
+    fail::cfg_callback("before-set-lock-in-memory", move || {
+        thread::sleep(Duration::from_millis(200));
```
Comment (you06) [inline tests/failpoints/cases/test_transaction.rs:77]: What's the purpose of this sleep before sending to channel?

Comment (sticnarf) [reply tests/failpoints/cases/test_transaction.rs]: It's now unncessary. I've removed it.

**Review by you06 — APPROVED** [review]

**Review by longfangsong — APPROVED** [review]

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**